### PR TITLE
Remove: Sync GitHub API removed

### PR DESCRIPTION
**Pull request by y0urself**
## What

* Sync GitHub API removed
* Updated dependencies

## Why

* deprecated through Async API

## References

DEVOPS-553
DEVOPS-545

## Checklist

- [ ] Tests

### CODE REVIEW

File: pontos/release/sign.py [hunk 1]
```diff
@@ -71,27 +71,27 @@ def sign(
...
     zip_destination = Path(f"{project}-{release_version}.zip")
-    with github.download_release_zip(
+    with github.releases.download_release_zip(
         repo, git_version, zip_destination
     ) as zip_progress:
         terminal.download_progress(zip_progress)
 
     tarball_destination = Path(f"{project}-{release_version}.tar.gz")
-    with github.download_release_tarball(
+    with github.releases.download_release_tarball(
```
Comment (y0urself) [inline pontos/release/sign.py:87]: same

File: pontos/release/sign.py [hunk 2]
```diff
@@ -71,27 +71,27 @@ def sign(
...
         repo, git_version, tarball_destination
     ) as tar_progress:
         terminal.download_progress(tar_progress)
 
     file_paths = [zip_destination, tarball_destination]
 
-    for progress in github.download_release_assets(repo, git_version):
+    for progress in github.releases.download_release_assets(repo=repo, tag=git_version):
```
Comment (y0urself) [inline pontos/release/sign.py:94]: here

**Review by y0urself — COMMENTED** [review]
@bjoernricks can you please fix the linting/test errors, regarding the github API? We still have used the sync API in `pontos-release`.

File: pontos/github/script/load.py [hunk 3]
```diff
@@ -98,10 +98,6 @@ async def run_async() -> int:
                 loop.close()
         return retval
 
-    # it's sync
```
Comment (bjoernricks) [inline pontos/github/script/load.py:101]: Shouldn't we raise an error here or just remove the if clause?

File: pontos/release/sign.py [hunk 4]
```diff
@@ -78,27 +78,27 @@ def sign(
     git_version: str = f"{git_tag_prefix}{release_version}"
     repo = f"{space}/{project}"
 
-    github = GitHubRESTApi(token=token)
+    github = GitHubAsyncRESTApi(token=token)
 
-    if not github.release_exists(repo, git_version):
+    if not github.releases.exists(repo=repo, tag=git_version):
```
Comment (bjoernricks) [inline pontos/release/sign.py:83]:
```suggestion
    if not await github.releases.exists(repo=repo, tag=git_version):
```

File: pontos/release/sign.py [hunk 5]
```diff
@@ -78,27 +78,27 @@ def sign(
...
     zip_destination = Path(f"{project}-{release_version}.zip")
-    with github.download_release_zip(
+    with github.releases.download_release_zip(
         repo, git_version, zip_destination
     ) as zip_progress:
         terminal.download_progress(zip_progress)
 
     tarball_destination = Path(f"{project}-{release_version}.tar.gz")
-    with github.download_release_tarball(
+    with github.releases.download_release_tarball(
```
Comment (bjoernricks) [inline pontos/release/sign.py:94]:
```suggestion
    async with github.releases.download_release_tarball(
```

File: pontos/release/sign.py [hunk 6]
```diff
@@ -78,27 +78,27 @@ def sign(
...
         repo, git_version, tarball_destination
     ) as tar_progress:
         terminal.download_progress(tar_progress)
 
     file_paths = [zip_destination, tarball_destination]
 
-    for progress in github.download_release_assets(repo, git_version):
+    for progress in github.releases.download_release_assets(repo=repo, tag=git_version):
```
Comment (bjoernricks) [inline pontos/release/sign.py:101]:
```suggestion
    async for progress in github.releases.download_release_assets(repo=repo, tag=git_version):
```

File: pontos/release/sign.py [hunk 7]
```diff
@@ -128,8 +128,8 @@ def sign(
     terminal.info(f"Uploading assets: {[str(p[0]) for p in upload_files]}")
 
     try:
-        for uploaded_file in github.upload_release_assets(
-            repo, git_version, upload_files
+        for uploaded_file in github.releases.upload_release_assets(
```
Comment (bjoernricks) [inline pontos/release/sign.py:131]:
```suggestion
        async for uploaded_file in github.releases.upload_release_assets(
```

File: pontos/release/sign.py (hunk 6 above)
Comment (bjoernricks) [reply pontos/release/sign.py:101]: I am really not sure if the API  stayed the same with async or if i refactored the code in some way.

**Review by bjoernricks — DISMISSED** [review]

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**